src/main/java/com/minecolonies/coremod/colony/CitizenData.java

```diff
@@ -145,13 +145,20 @@ public void readFromNBT(@NotNull final NBTTagCompound compound)
     }
 
     /**
-     * return the entity instance of the citizen data.
+     * Return the entity instance of the citizen data. Respawn the citizen if needed.
      *
      * @return {@link EntityCitizen} of the citizen data.
      */
+    //TODO evaluate if this is nullable. Will spawnCitizen fail?
     @Nullable
```

Comment:
We could improve this whith the respawn so that we immediately return the spawned citizen and remove the null. Would save us quite a few checks everywhere